src/python/pants/engine/environment.py

```diff
@@ -17,16 +17,23 @@
 name_value_re = re.compile(r"([A-Za-z_]\w*)=(.*)")
 shorthand_re = re.compile(r"([A-Za-z_]\w*)")
 
+
 """
-Accesses to `os.environ` cannot be accurately tracked, so @rules that need access to the
-environment should use APIs from this module instead.
+TODO: This module currently contains two concepts:
+
+1. environment variable matching (CompleteEnvironment, Environment, EnvironmentRequest)
+2. the new "environment" concept from #13682.
 
-Wherever possible, the `Environment` type should be consumed rather than the
-`CompleteEnvironment`, as it represents a filtered/relevant subset of the environment, rather
-than the entire unfiltered environment.
+If the second use case gains steam as #13682 proceeds, we should move and rename the
+environment-variable matching APIs to remove ambiguity.
 """
 
 
+@dataclass(frozen=True)
+class EnvironmentName:
+    """TODO: Replace with `pants.core.util_rules.environments.ResolvedEnvironmentAlias`."""
```

Comment:
This seems worth figuring out before merging

Reply:
Possibly? I think that it would probably be good as followup, because a lot of the product_request callsites which create `EnvironmentName()` on the fly right now will need to be updated to use a bootstrapped environment, and that might be a bit of work on its own.

This change is primarily about the interface change.

Reply:
But the import path and name might change? For example, the option is called `aliases`, not `names`. I don't see why we would define this in `engine/` rather than `core/util_rules/` also - import cycle?

Bump that it seems like the `Platform` changes can be split out as a dedicated PR.